exe-common/src/network/ntt.rs

```diff
@@ -0,0 +1,73 @@
+use futures::Future;
+use tokio::runtime::Runtime;
+
+use network::api::{Api, BlockRef};
+use network::{Error, Result};
+
+//to_socket_addr
+use network_core::client::block::HeaderService;
+use network_ntt::client as ntt;
+use std::net::SocketAddr;
+
+use cardano::{
+    block::{Block, BlockHeader, HeaderHash, RawBlock},
+    tx::{TxAux, TxId},
+};
+
+pub struct NetworkCore {
+    handle: ntt::ClientHandle<Block, TxId>,
+    pub rt: Runtime,
+}
+
+impl NetworkCore {
+    pub fn new(sockaddr: SocketAddr) -> Result<Self> {
+        trace!("New network core: {}", sockaddr);
+        let connecting = ntt::connect(sockaddr);
+        match connecting.wait() {
+            Ok((connection, handle)) => {
+                // FIXME: use default executor, or take
+                // executor argument before merge.
+                let mut rt = Runtime::new().unwrap();
+                rt.spawn(connection.map(|_| {
+                    println!("Exited");
+                }));
+                Ok(NetworkCore { handle, rt })
+            }
+            Err(_err) => unimplemented!(),
+        }
+    }
+}
+
+impl Api for NetworkCore {
```

Comment:
Ugh, why do we have this trait named `Api` to begin with. Not a problem of this PR, of course.

Reply:
old exe-common things...